cmd/gin-repod/repo.go

```diff
@@ -913,3 +913,50 @@ func (s *Server) repoDescription(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	w.Write(body)
 }
+
+// listRepoCommits returns a list of all commits from the branch of a specified repository as json.
+// Required access level is PullAccess.
+func (s *Server) listRepoCommits(w http.ResponseWriter, r *http.Request) {
+
+	ivars := mux.Vars(r)
+	rid, err := s.varsToRepoID(ivars)
+
+	ibranch := ivars["branch"]
+
+	_, ok := s.checkAccess(w, r, rid, store.PullAccess)
+	if !ok {
+		return
+	}
+
+	repo, err := s.repos.OpenGitRepo(rid)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	ok, err = repo.BranchExists(ibranch)
```

Comment:
Maybe create something like `RefExists` because the rest of the code could easily work for tags too.